Code/Legacy/CrySystem/SystemInit.cpp

```diff
@@ -352,39 +352,31 @@ bool CSystem::InitFileSystem_LoadEngineFolders(const SSystemInitParams&)
 //////////////////////////////////////////////////////////////////////////
 bool CSystem::InitAudioSystem(const SSystemInitParams& initParams)
 {
-    if (!Audio::Gem::AudioSystemGemRequestBus::HasHandlers())
+    if (!Audio::Gem::SystemRequestBus::HasHandlers())
     {
-        // AudioSystem Gem has not been enabled for this project.
+        // AudioSystem Gem has not been enabled for this project/configuration.
         // This should not generate an error, but calling scope will warn.
         return false;
     }
 
-    bool useRealAudioSystem = false;
-    if (!initParams.bPreview
-        && !m_bDedicatedServer
-        && m_sys_audio_disable->GetIVal() == 0)
-    {
-        useRealAudioSystem = true;
-    }
-
     bool result = false;
-    if (useRealAudioSystem)
-    {
-        Audio::Gem::AudioSystemGemRequestBus::BroadcastResult(result, &Audio::Gem::AudioSystemGemRequestBus::Events::Initialize, &initParams);
-    }
-    else
-    {
-        Audio::Gem::AudioSystemGemRequestBus::BroadcastResult(result, &Audio::Gem::AudioSystemGemRequestBus::Events::Initialize, nullptr);
-    }
 
-    if (result)
+    bool shouldInitializeAudioSystem = (!initParams.bPreview && !initParams.bUnattendedMode);
```

Comment:
Why not just always initialize this in the SystemComponent of the Audio Gem and use a settings registry entry for the determining if the audio system should initialize.

The bPreview and bUanttendedMode parameters seem unrelated to Audio system and I think it would be better if the Audio system had an explicit option.

Afterwards we can just strip out this code.

Reply:
Removing these dependencies is what I was hoping to simplify this to.  I was looking into these options and couldn't tell for sure whether these options are being used.

`bPreview` mode appears to preview files.  But it only applies to legacy file types "chr", "skin", "cdf", "cgf", ... (see `IsPreviewableFileType`).  I can probably remove the dependency on that option.
`bUnattendedMode` appears to be set when exporting.  Do we still use export levels?  I don't think so.

If these appear to be useless, I'll remove them and simplify this.

Reply:
No we don't export levels at all anymore.
For the old AZ::IO::Archive format, we create the PAK via the zip command to compress a levels files.
`bPreview` appears useless.

Also even if those legacy were being previewed in some kind of special Editor mode, what harm would their be to have the Audio system active?

Reply:
I think the intention was that disabling full audio init trims the fat in these modes.  Harmless if it's there, but it's unnecessary.

I'll remove this stuff.